### PR TITLE
Update bundled plugins after 2022-05-17 security advisory

### PR DESCRIPTION
**Pull request by daniel-beck**
See https://www.jenkins.io/security/advisory/2022-05-17/#SECURITY-2604 and https://www.jenkins.io/security/advisory/2022-05-17/#SECURITY-2116

### Proposed changelog entries

* Update bundled Script Security Plugin from 1.75 to v35f6a_0b_8207e
* Update bundled WMI Windows Agents Plugin from 1.0 to 1.8.1

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

**Review by basil — COMMENTED** [review]
This PR causes `hudson.model.AbstractProjectTest#configDotXmlSubmissionToDifferentType` to start failing with:

```
java.io.IOException: Failed to install script-security plugin
	at hudson.PluginManager.dynamicLoad(PluginManager.java:940)
	at hudson.PluginManager.dynamicLoad(PluginManager.java:875)
	at org.jvnet.hudson.test.UnitTestSupportingPluginManager.installDetachedPlugin(UnitTestSupportingPluginManager.java:170)
	at hudson.model.AbstractProjectTest.configDotXmlSubmissionToDifferentType(AbstractProjectTest.java:442)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:607)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.IOException: Failed to load: Script Security Plugin (script-security 1172.v35f6a_0b_8207e)
 - Plugin is missing: caffeine-api (2.9.2-29.v717aac953ff3)
	at hudson.PluginWrapper.resolvePluginDependencies(PluginWrapper.java:994)
	at hudson.PluginManager.dynamicLoad(PluginManager.java:926)
	... 16 more
```

Since `matrix-project` is already on the test classpath via `test/pom.xml`, I see no reason to install it via the detached plugin mechanism as that test is doing. Deleting all calls to `TestPluginManager#installDetachedPlugin` from that test (and therefore the installation of `matrix-project` through the detached plugin mechanism) should not affect the correctness or coverage of that test (because `matrix-project` is already present on the test classpath via `test/pom.xml`), but it should resolve the test failure in this PR as well as making the test less fragile with regard to future changes.

**Review by res0nance — APPROVED** [review]

File: test/src/test/java/hudson/model/AbstractProjectTest.java
Comment (basil) [inline test/src/test/java/hudson/model/AbstractProjectTest.java:437]: This variable is now unused. Unused variables are generally considered bad programming practice by most style guides, so I suggest deleting it (along with the corresponding import):

```suggestion
```

This makes the code easier to read by avoiding an unused variable.

**Review by basil — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
